irodsConnector/session.py

```diff
@@ -15,28 +15,64 @@
 
 class Session(object):
     """Irods session operations """
-    _session = None
-    context = utils.context.Context()
 
-    def __init__(self, password=''):
+    def __init__(self, context: utils.context.Context, password=None, greet=False):
         """ iRODS authentication with Python client.
 
         Parameters
         ----------
         password : str
             Plain text password.
+        context : utils.context Context object containing valid
+            irods environment and ibridges configuration 
+        greet: bool
+            Print some greeting and irods info to prompt
 
-        The 'password' property can autoload from its cache, but can be
+        The 'password' is autoloaded from its cache, but can be
         overridden by `password` argument.  The iRODS authentication
         file is expected in the standard location (~/.irods/.irodsA) or
         to be specified in the local environment with the
         IRODS_AUTHENTICATION_FILE variable.
 
         """
         self._password = password
+        self._ienv = context.irods_environment.config
+        self._conf = context.ibridges_configuration.config
```

Comment:
Firstly, if either of these `JsonConfig`s fail to load for any reason, there will be an error something like:

```python
AttributeError: NoneType object has no attribute 'config'
```

Secondly, there is no lazy instantiation benefit here.  The instant you access the `irods_environment` or `ibridges_configuration` property of the **context**, and the `config` property for each of those, you are first instantiating `JsonConfig` objects, then loading their dictionaries from disk.  When this is done as an iBridges `Session` instance is initialized, these four instances are created, and only at that point.  Resetting them after a 'Close session' becomes more cumbersome as a result, requiring extra manual methods.  The `@property`s do this naturally as part of their design.